### ISSUE TITLE
The manual is not printable

### ISSUE THREAD

**Issue by Atcold** [pgf-tikz/pgf#1183]
### Version

3.1.9a

### Details

Trying to print the manual using Adobe Acrobat Pro DC `2022.001.20169` gives the following errors.

> The document could not be printed.

and

> There were no pages selected to print.

![image](https://user-images.githubusercontent.com/2119355/183470409-b23bc2e9-249a-49b7-b104-9a117bf6c6d3.png)
![image](https://user-images.githubusercontent.com/2119355/183470458-eb4574de-cc3d-4211-a7f4-508c13db3943.png)

**Comment by Atcold** [comment]
Can anyone confirm or dispute this issue?

**Comment by Atcold** [comment]
Can anyone help?

**Comment by salewski** [comment]
Hi @Atcold, I am able to print the manual without any problem, though due to its length I only tested by printing specific pages. I tested printing via the `evince(1)` PDF viewer and via `lpr(1)`:

```console
    $ lpr -o page-ranges=2 pgfmanual.pdf
```

I do not have `Adobe Acrobat Pro DC` to test with, so cannot speak to any experience with that specific program.

Are you having difficulty printing only from `Adobe Acrobat Pro DC`? Do you have other programs that could try?

For completeness, I'll note that my tests were performed using the PDF of the manual downloaded today (`2024-10-27`) via:

```console
$ curl -L -O -R 'https://pgf-tikz.github.io/pgf/pgfmanual.pdf'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 9219k  100 9219k    0     0  18.3M      0 --:--:-- --:--:-- --:--:-- 18.2M
```

In case it's useful now or in the future, here's some additional metadata for the file:
```console
$ stat --format='%s:  %n' pgfmanual.pdf
9440725:  pgfmanual.pdf

$ file pgfmanual.pdf
pgfmanual.pdf: PDF document, version 1.5

$ pdfinfo pgfmanual.pdf
Title:
Subject:
Keywords:
Author:
Creator:        LaTeX with hyperref
Producer:       LuaTeX-1.18.0
CreationDate:   Thu Oct 24 13:08:56 2024 EDT
ModDate:        Thu Oct 24 13:08:56 2024 EDT
Tagged:         no
UserProperties: no
Suspects:       no
Form:           none
JavaScript:     no
Pages:          1322
Encrypted:      no
Page size:      595.276 x 841.89 pts (A4)
Page rot:       0
File size:      9440725 bytes
Optimized:      no
PDF version:    1.5
```

**Comment by Atcold** [comment]
I wanted to use Adobe Acrobat because I wanted to print it as a booklet (4 pages per page with correct front and back).
Other readers, such as Mac's Preview, don't display its content correctly.
For example, here's page 166 of `pgfplots` (which showcases the limitations of Preview in displaying Ti*k*Z based drawings).

![image](https://github.com/user-attachments/assets/e39d7eb5-1e6a-40e0-8a38-b4c39190078c)

Does Evince display this page correctly **and** can it print a PDF as a booklet?

**Comment by jasperhabicht** [comment]
Can you try to find out which page is the problematic one? I could imagine that you can print some pages with Acrobat, but only one or some pages are problematic. 

If you know which page has the issue, it will be easier to test and to identify the reason for this issue.

**Comment by Atcold** [comment]
Thanks, @jasperhabicht, for the suggestion.
*Any* page will throw the error I posted above.

**Comment by salewski** [comment]
> I wanted to use Adobe Acrobat because I wanted to print it as a booklet (4 pages per page with correct front and back). Other readers, such as Mac's Preview, don't display its content correctly. For example, here's page 166 of `pgfplots` (which showcases the limitations of Preview in displaying Ti_k_Z based drawings).
> [...]
> 
> Does Evince display this page correctly **and** can it print a PDF as a booklet?

No, `evince(1)` does not display that page correctly, either. On my system, the `pgfplots.pdf` file is `/usr/share/texlive/texmf-dist/doc/latex/pgfplots/pgfplots.pdf`, from the `texlive-pictures-doc` Debian package (version `2020.20210202-3`), and those `tikzpicture` examples are on pages 163 and 164:

![screenshot-pgfplots pdf--pg-163--evince](https://github.com/user-attachments/assets/07735f67-9c43-4c88-af2a-813b09270dfe)

I've confirmed that printing them from evince also results in the blurred contour levels, too. (Near the top of page 163 is a "Viewer Limitations" warning that says some viewers will not render the examples correctly, and explicitly names `evince` as one of the viewers known to not work with those specific examples.)

However, viewing the document in `xpdf(1)` *does* show the expected output:

![screenshot-pgfplots pdf--pg-163--xpdf](https://github.com/user-attachments/assets/96592960-4f0a-4b35-86fc-0551859b27f0)

And printing those specific pages using `lpr(1)` (from CUPS) works:

```console
$ lpr -o number-up=4 -o page-ranges=41 /usr/share/texlive/texmf-dist/doc/latex/pgfplots/pgfplots.pdf
```
(Note the calculation of the page range -- `164 / 4 = 41`, which means that logical page 164 will be the highest-numbered page emitted on the 4-up output on hardcopy page 41. Thus the printed page will have logical pages 161-164.)

You might be able to use that approach to test printing specific pages, or groups of pages.

To get the double-sided printing you mentioned you want, you would need to add the `-o sides=two-sided-long-edge` option:

```console
$ lpr -o number-up=4 -o sides=two-sided-long-edge -o page-ranges=41 /path/to/pgfplots.pdf
```

Adjust the `-o page-ranges` option to suit (or omit it to print the entire doc).

I'm not sure that the above will give you the output you're seeking, however, since the `pgfplots.pdf` PDF itself does not have blank pages inserted into it to (for example) force chapters to start on an even-numbered page. Is that sort of thing important for the boolket you're trying to print? Or are you just trying to get 4-up printing, front and back? The above would give you the latter...

**Comment by Atcold** [comment]
Yes, the page ordering is essential if I plan to cut the printed pages in half and put together an actual book.

But the way, I've started printing some section of the full page (so, I gave up saving paper).

Page 180 does not print.
It successfully goes to the printer (when I send it through Preview) but the printer does absolutely nothing.

<img width="823" alt="image" src="https://github.com/user-attachments/assets/3e1f8374-3350-460d-ab88-abb51b0874df">

**Comment by Atcold** [comment]
Same with page 188.

<img width="822" alt="image" src="https://github.com/user-attachments/assets/1dea4b44-90c0-4df2-a83c-45b20bc4c445">

If I try printing chapter 15, the printer consistently stops at page 179.
I tried printing explicitly 180. Nothing happens.
I tried printing the rest of the chapter. The printer stops at 187.

**Comment by Atcold** [comment]
Page 202 is non-printable as well.

<img width="823" alt="image" src="https://github.com/user-attachments/assets/eb487644-cd0f-48e4-a1ff-698abd34b6da">

**Comment by Atcold** [comment]
203.

<img width="821" alt="image" src="https://github.com/user-attachments/assets/41cbe494-605d-40a0-b2cb-c3da74e7cbab">

**Comment by Atcold** [comment]
353.

<img width="822" alt="image" src="https://github.com/user-attachments/assets/22bc3d41-0203-44c2-8141-e3a1db616571">

**Comment by Atcold** [comment]
And basically every page after 359 of section 23.4 Fadings.
This makes me assume that fadings are non-printable.

**Comment by Atcold** [comment]
@jasperhabicht: I've isolated several problematic pages. See above.
@salewski: thanks for your input. Let me address your points.

There's a difference between booklet layout printing 2 pages per paper and sequential printing of 2 pages per paper.
Say there are 4 pages in total.
The booklet ordering will be:
```
┌──────────────┐  ┌──────────────┐
│┌─────┐┌─────┐│  │┌─────┐┌─────┐│
││     ││     ││  ││     ││     ││
││  4  ││  1  ││  ││  2  ││  3  ││
││     ││     ││  ││     ││     ││
│└─────┘└─────┘│  │└─────┘└─────┘│
└──────────────┘  └──────────────┘
```
So that, after cutting, page 1 is followed by page 2 and page 3 is followed by page 4.
Sequential printing would give you:
```
┌──────────────┐  ┌──────────────┐
│┌─────┐┌─────┐│  │┌─────┐┌─────┐│
││     ││     ││  ││     ││     ││
││  1  ││  2  ││  ││  3  ││  4  ││
││     ││     ││  ││     ││     ││
│└─────┘└─────┘│  │└─────┘└─────┘│
└──────────────┘  └──────────────┘
```
And after catting, page 2 is followed by page 3 and page 4 is followed by page 1.
Anyway, I moved forward and printed one page per paper side (front and back).

Can you see whether `lpr` works for any of the pages I posted above?
My printer receives the page and does absolutely nothing.

**Comment by Atcold** [comment]
Anyone?

**Comment by salewski** [comment]
> 
> Can you see whether `lpr` works for any of the pages I posted above? My printer receives the page and does absolutely nothing.

@Atcold : I'm not able to test anything just now, but I'll check the `lpr` behavior on more pages this weekend and share my findings.

**Comment by muzimuzhi** [comment]
I don't have access to printers, but you could try to repair(?) the manual using PDF tools like [`mutool clean`](https://mupdf.readthedocs.io/en/latest/mutool-clean.html).

**Comment by salewski** [comment]
> > Can you see whether `lpr` works for any of the pages I posted above? My printer receives the page and does absolutely nothing.
> 
> @Atcold : I'm not able to test anything just now, but I'll check the `lpr` behavior on more pages this weekend and share my findings.

@Atcold : I was able to successfully print three of the pages from `pgfmanual.pdf` that you above noted were not printing for you: `180`, `188`, and `202`.

I stopped trying after those three since I'm just killing trees over here, and we're clearly getting different results.

FWIW, the printer I used for testing is an HP Color LaserJet Pro M283fdw MFP, circa 2019.

**Comment by jasperhabicht** [comment]
Just a side note: It should be possible to print the pages when you tell the computer to rasterize the ouput before sending it to the printer. I think, Adobe Reader has some option "Advanced" > "Print as Image".
